### ISSUE TITLE
YouTube Unable to Playback after 1 minute / Unable to Playback after Reinstall

### ISSUE THREAD

**Issue by dsoto8** [futo-org/grayjay-android#1596]
### What happened?

Good Morning, Grayjay Community!

I have encountered an issue wherein the Grayjay application fails to play YouTube content after 1 minute of playback. (Initially, YouTube content had stopped playback entirely, requesting login credentials; after supplying my information, this was the result.)

I attempted reinstalling Grayjay; now, it refuses to play any YouTube content, even after supplying credentials.

Any assistance would be greatly appreciated.

Thanks!

• Samsung S22 Ultra (SM-S908U1)
• Android v14
• OneUI v6.1

### Grayjay Version

268

### What plugins are you seeing the problem on?

Youtube

### Plugin Version

218

### When do you experience the issue?

- [X] While logged in
- [X] While logged out
- [ ] N/A

**Comment by Zvonimir-FUTO** [comment]
Could you try enabling **Force Experimental UMP Streams** in Youtube plugin settings and check if that helps?

**Comment by dsoto8** [comment]
> Could you try enabling **Force Experimental UMP Streams** in Youtube plugin settings and check if that helps?

Hi, Zvonimir-FUTO!

Thanks for the suggestion!

Just attempted; app behavior is unchanged.

**Comment by Cederick** [comment]
I'm having the same issue since today

**Comment by Cederick** [comment]
Enabling the option to prefer webm audio and video in the settings fixed it for me for some reason.

**Comment by dsoto8** [comment]
Thanks for the additional suggestions!

Enabling prefer webm audio failed to resolve the issue for me.

**Comment by Zvonimir-FUTO** [comment]
We are having issues reproducing this one so could you guys tell us from which country you are if you don't mind disclosing that information publicly.

Also if you have any idea how to replicate this it would be helpful. If you are using VPN, connected over Wi-Fi, mobile data, or anything you think might cause the issue.

**Comment by dsoto8** [comment]
> We are having issues reproducing this one so could you guys tell us from which country you are if you don't mind disclosing that information publicly.
> 
> Also if you have any idea how to replicate this it would be helpful. If you are using VPN, connected over Wi-Fi, mobile data, or anything you think might cause the issue.

Hi, Zvonimir-FUTO!

Thanks for the information!

I am in the U.S.A.; the issue occurs when connected to either WiFi or mobile data; I have a VPN enabled.

I have just tested YouTube video playback both with VPN enabled and disabled:  Disabling VPN resolves the issue, even restoring playback without requiring a login; re-enabling VPN causes the issues described to recur.

**Comment by Zvonimir-FUTO** [comment]
Thank you for the feedback @dsoto8. Unfortunately, this is out of our hands since this is Youtube blocking you on the bases of _suspicious_ activity because of the VPN.

>I had to disable Force Experimental UMP Streams as it was preventing YouTube videos from playing for me I'm not sure why.

As the name implies these are still experimental and shouldn't be used unless everything else fails.

**Comment by Zvonimir-FUTO** [comment]
> I'm not using a VPN and I have changed my IP multiple times. Also used my 5G and still have the same issue. I'm from Germany.

Apologies but I though you fixed the issue when you commented:

>Enabling the option to prefer webm audio and video in the settings fixed it for me for some reason.

**Comment by Cederick** [comment]
So everything seems to be working again so my best guess is that it was on YouTubes side. Now I'm only back to Spotify not playing audio via fcast and only for less than a second with Google cast but that's something for a new issue.
Sorry for causing the extra working for not retesting before replying.

**Comment by Zvonimir-FUTO** [comment]
No apologies needed :)

**Comment by Zvonimir-FUTO** [comment]
Think we found the cause of this issue and we released a fix in new Youtube update. Please update and keep us updated if the issue continues occurring

**Comment by Christopher-Hayes** [comment]
On YouTube version 219 I'm still seeing this issue. Regardless of whether I have "Force Experimental UMP Streams" enabled or not.

**Comment by SovietGP** [comment]
No video from will play for over 1min, doesn't matter if I change to webm or not, I tried playing around with every setting imaginable to no avail, ended up resetting to default, also Spotify doesn't play anything, everything is just frozen and with a play button that doesn't do anything.

**Comment by Zvonimir-FUTO** [comment]
> No video from will play for over 1min, doesn't matter if I change to webm or not, I tried playing around with every setting imaginable to no avail, ended up resetting to default, also Spotify doesn't play anything, everything is just frozen and with a play button that doesn't do anything.

just to confirm, you are not using VPN?

**Comment by Auster-1** [comment]
I have an issue that appears to be related, so sending it here:

While logged in, with VPN active (but not hopping around), and on either stream or download, the video fails early on.
On stream, as in the original post, the video gets to around 1 minute, then stops loading, and becomes unresponsive to the Grayjay player until the video is reopened, and then proceeds to have the same issue.
On download, it shows the video size as a few dozen to a few hundred bytes, no matter the size of the video, and throws the following error:
`[Youtube] No streams for requesting segment 7 (2384) [3-4]`

Active: allow loginfallback, allow age restricted, allow controversial restricted

Also tried just now, but in separated tests (neither worked): activating force experimental ump streams, disabling VPN

Youtube extension version 219, Grayjay version 278

**Comment by kick82** [comment]
So, I'm having exactly the same issue 
App ver. is 278.
Plugin is 219.
Mullvad VPN. 
Without VPN everything works ok. 
With VPN app says failing to get iOS stream, fallback to UMP and then freezes after 1 minute of playback.
Problem is that without VPN there are other problems related to youtube restrictions in my country, so, have to use VPN.

**Comment by Zvonimir-FUTO** [comment]
Related to #1610

**Comment by Ryan52003** [comment]
I've just now started having the same issue, any fixes?

**Comment by SovietGP** [comment]
> I've just now started having the same issue, any fixes?

turninf off VPN 8f you uave one, otherwise bo. I've been having this issue for over a month and no fixes yet

**Comment by Ryan52003** [comment]
> > I've just now started having the same issue, any fixes?
> 
> turning off VPN if you have one, otherwise no. I've been having this issue for over a month and no fixes yet

Ok well now it somewhat fixed itself, but i'm only getting this issue on age restricted videos. I've used grayjay for months without any issues until I got it this morning

**Comment by Gizmo144** [comment]
Experiencing something similar. Latest app and plugin versions.

Enabling webm preference does nothing.

However, there is a slight difference for me. The 1min issue only occurs when I enable force UMP and will not go further. It errors out with "could not load UMP stream".

When UMP is disabled then all videos go through the "block detected, attempting bypass" infinite loop. This ONLY occurs on WiFi. Videos load fine on mobile data. Note, this is any WiFi whether there is a VPN or not.

One additional message I always see when ever I try play a video I will always receive a message about using iOS sources fallback (cached). This appears whether I get the block message or not. 

Country: South Africa
Plugin version: 219
App version: 278
Android: 14
One UI: 6.1
Phone: Samsung Galaxy S24 Ultra

**Comment by K4TieK** [comment]
Im pretty much experiencing this issue now too
On my wifi i get the issues
"Failed to get IOS data, falling back to UMP"(idk what that is tbh)
Mobile data, video works fine
Vpn on, video works fine
Vpn off and on home wifi, video stops after 60 seconds 


Phone:Samsung a71
Country:United Kingdom 
Grayjay version:278
Youtube plugin version:219
Plus im using the Test version of the Youtube plugin too, version 206

Im logged into both plugins

**Comment by anonobunch** [comment]
totally broken.

**Comment by thecatontheceiling** [comment]
I have started getting the same issues today :/

**Comment by jonathancutting** [comment]
I am also having this issue with YouTube now.

This thread and issue #1610 may be related.

This just started (for me) about three days ago.
- All YouTube videos fallback to UMP.
- If watching video on phone, video playback stops after about 1 minute.
- If streaming via fcast, videos won't play at all.
- Half the videos in fcast will "load" forever; the other half throw server errors and force fcast to repeatedly disconnect and reconnect forever.
- Problem persists with every permutation of connected and not connected to VPN on phone and fcast device.
- Native YouTube app plays videos.
- Odysee and Rumble seem to work fine, just YouTube is broken (although Odysee suffers from other, untelated issues).

**Comment by dankxylese** [comment]
Having the same issue over wifi in the UK, no VPN. Interestingly enough NewPipe works fine on all effected videos with default settings. Maybe look into adding their implementation as a workaround?

**Comment by TypicalScenario** [comment]
Hi! I'm also experiencing the UMP fallback with 1 minutes issue. Region is middle of EU, app version is 278, youtube plugin version is 219, logged in. Other plugins aren't affected, only Youtube.

A couple things that might tip off the team:

- It started with me trying to watch 2 age restricted videos back to back. Trying, because the issues started on one of them.
- I noticed around the time, that then the plugin showed that I was logged out, which wasn't true because after tapping on login, I just clicked on my channel and didn't have to log into the account at all, the cookie was still there.
- After a complete log-out and relog-ing, the first video didn't fall back to UMP and played normally, but from the next onward the problem came back.

**Comment by Zvonimir-FUTO** [comment]
Hey guys, thank you for the feedback but issue has already been identified and we are working on a fix. There has been quite a progress on it but each time we think we finished something new pops up so not quite done yet. Anyway, this is top priority and we will release a fix as soon as it is ready.

**Comment by Christopher-Hayes** [comment]
I'll add that even when I go to YouTube.com on my mobile browser, videos no longer play in my mobile browser since a few days ago, it just endlessly buffers. Only way I got videos to play was by opening an incognito tab, and logging into YouTube from there.

On desktop, YouTube.com in the browser will only play videos, but ONLY if I'm logged in. GrayJay on desktop has issues similar to mobile for me.

So, Google is definitely making another big effort.

**Comment by thecatontheceiling** [comment]
Does Google not realize that the kind of people using Grayjay are _determined_ and will literally never switch to using the official app? Or maybe it's just me. I despise the official app btw

**Comment by NashadkaJr** [comment]
> I'll add that even when I go to YouTube.com on my mobile browser, videos no longer play in my mobile browser since a few days ago, it just endlessly buffers. Only way I got videos to play was by opening an incognito tab, and logging into YouTube from there.
> 
> On desktop, YouTube.com in the browser will play videos, but ONLY if I'm logged in. GrayJay on desktop has issues similar to mobile for me.
> 
> So, Google is definitely making another big effort.

yt-dlp also has stopped working for me. Google's absolutely cracking down on people watching YouTube without their official client.

**Comment by Zvonimir-FUTO** [comment]
Hi everyone, this has been finally fixed and new version of Grayjay and Youtube have been released!

Go update and test and tell us if you find issues except the one below.

We are aware that age restricted, controversial videos, or other videos that require login may not yet work. We will be working on this next week.

The desktop release that is required for this fix will also be released next week.

**Comment by barkerd427** [comment]
I updated the app, but it still isn't working. I am getting a different error, though.

![Image](https://github.com/user-attachments/assets/f6b1bb10-b717-48a5-a104-358f8408a30b)

**Comment by neverson42** [comment]
I just installed the update. Things were working great for a few vids until Brandon Herrera's latest video (a gun meme review) came up in my queue. I thought it might be a restricted video (due to nature of the content), but im not getting forced to log in when attempting to view in a browser that I'm not logged in on. Might still be some issues.

**Comment by Zvonimir-FUTO** [comment]
can you guys try turning off **Use Login for video details** if it is turned on?
Can you confirm you are on 281?
On which Youtube version are you?
Can you check if you are getting iOS streams? (select video resolution and it should say)
Can you enable Force Experimental UMP Streams in the Youtube settings?
Are you logged in or not?

**Comment by neverson42** [comment]
I just logged out of the app and back in to check if that could be an issue, so it's possible that might have been an issue.

I am on app ver 281 with yt plugin version 221.

I did NOT have "Use Login for video details" enabled. After turning that on (and relogging into youtube - again, not sure if that did anything), things are working. TBD if they stay this way. I'll let you know if the problems come back.

Thank you!

**Comment by Zvonimir-FUTO** [comment]
Hey thank you for the feedback. Maybe this solved the issue for other.

**Comment by barkerd427** [comment]
I do not have **Use Login for video details** turned on.
I am on 281.
YouTube version is 221.
I am only getting UMP now, but earlier today I could switch between them.
I had the Force Experimental UMP Streams on and it wasn't working. I also tried it with the prefer webm setting on, and with all of them off. No version is working.
I am logged into YT. I also logged out and tested, and then logged back in and tested.

I have seen the first failure I get is No streams for requesting segment 7, and then the rest are the same as I posted in the image. I am also adding a new download each time to ensure there is one that isn't effected by something done previously.

**Comment by Zvonimir-FUTO** [comment]
Will have to continue investigation tomorrow. Thank you for your time and apologies for inconvenience.

**Comment by barkerd427** [comment]
Thank you for working on this. The app has been great otherwise! I will try to watch my messages if you need me to test something, and I'll let you know if I get it to work otherwise.

**Comment by neverson42** [comment]
It started happening again on my drive home from work. Looks like y'all are on it already. Thanks!

**Comment by LJmenace** [comment]
Got the update still happening. But now when downloading I get the following error. Some videos don't play at all others just the very beginning. 

* Cannot get properties of undefined length.

![Image](https://github.com/user-attachments/assets/1a22dfd2-9d08-4aaa-8b7d-ecef57249603)

**Comment by LJmenace** [comment]
Update: Works now.

Turned on the following on YouTube plugin.

![Image](https://github.com/user-attachments/assets/c5670756-93af-4b49-8a36-97aa6c3345eb)

I'm able to watch without any issues and download as well.

![Image](https://github.com/user-attachments/assets/0e4234d1-2e35-434a-8512-8eb7a42b1cc9)

**Comment by thecatontheceiling** [comment]
I am still having this issue after the update 

I messed with a few settings in the YouTube source but nothing seemed to be doing anything, I noticed it was now saying "Signature invalid" or something like that, not sure what that is about

**Comment by thecatontheceiling** [comment]
> I do not have **Use Login for video details** turned on.
> I am on 281.
> YouTube version is 221.
> I am only getting UMP now, but earlier today I could switch between them.
> I had the Force Experimental UMP Streams on and it wasn't working. I also tried it with the prefer webm setting on, and with all of them off. No version is working.
> I am logged into YT. I also logged out and tested, and then logged back in and tested.
> 
> I have seen the first failure I get is No streams for requesting segment 7, and then the rest are the same as I posted in the image. I am also adding a new download each time to ensure there is one that isn't effected by something done previously.

Following this seems to have fixed the issue for me, thank you :)

**Comment by thecatontheceiling** [comment]
Apparently very temporarily. Half an hour later I'm constantly getting blocked by YT. Thanks google

**Comment by AndrewHack242** [comment]
Updated and still having the issue

**Comment by LJmenace** [comment]
Guess it was just a one time fluke. Now I'm also having issues. Changing settings in YouTube plugin doesn't help. Most videos are back to only playing the first few secs

**Comment by cwol97401** [comment]
Two updates ago this problem started for me. I noticed then that webm video format stopped appearing and only options were ump or iOS. When I get a video with ios formatting, it plays thru, but ump does not. I can click on a video and it first only offers ump but if I back out and reload the same video, there's a 20% chance that I'll land the iOS version and all is good. Don't know if this helps at all.

**Comment by getabc** [comment]
Same issue in Canada. 

I found that if the video has a mp4 and webm source available for download, I can download and play the media, but if it only has webm source it would freeze and fail downloads.

**Comment by Fogapod** [comment]
reopening video 3-4 times until UMP sourcrs stop appearing works for me. app restarts, logouts or editing settings don't help

**Comment by barkerd427** [comment]
> reopening video 3-4 times until UMP sourcrs stop appearing works for me. app restarts, logouts or editing settings don't help

This is the answer for me. If you open the download prompt repeatedly then eventually the IOS sources appear and they have worked for every video.

**Comment by PGQT** [comment]
It would be wonderful to be able to disable UMP sources altogether for moments like these where it just breaks stuff.

**Comment by Queuecumber** [comment]
> This is the answer for me. If you open the download prompt repeatedly then eventually the IOS sources appear and they have worked for every video.

This is also "100%" reliable for me, if I open the same video enough times it eventually loads the iOS streams and plays normally (I hope this is a hint as the the problem)

**Comment by thecatontheceiling** [comment]
That doesn't work for me

**Comment by Zvonimir-FUTO** [comment]
> It would be wonderful to be able to disable UMP sources altogether for moments like these where it just breaks stuff.

Well UMP sources are the last thing we try but Youtube pushes them on the users so there is nothing we can do about it then adapt.

**Comment by MikeL270** [comment]
> can you guys try turning off **Use Login for video details** if it is turned on?
> Can you confirm you are on 281?
> On which Youtube version are you?
> Can you check if you are getting iOS streams? (select video resolution and it should say)
> Can you enable Force Experimental UMP Streams in the Youtube settings?
> Are you logged in or not?

Logging out fixed it for me.

**Comment by jonathancutting** [comment]
GrayJay version: 281
YouTube version: 221
Use login for video details: OFF
iOS streams: sometimes, seems random, mostly UMP
Enable force UMP: ON
Logged into YouTube: YES

Getting the issues discussed above, and these issues extend to Fcast streams. Sometimes takes half a dozen stops and retries to get a video to play on phone or Fcast, and many stop after 1 minute. I get the "UMP fallback" message on almost every YouTube video. Logging out or connecting to a VPN makes it worse.

**Comment by VoxelPrismatic** [comment]
This bug is now occurring to me at 0:19s. instead of at one minute.

**Comment by MikeL270** [comment]
> This bug is now occurring to me at 0:19s. instead of at one minute.
> 
> I am using a VPN.

YouTube is pretty quick to blacklist VPN IP addresses in my experience.

**Comment by VoxelPrismatic** [comment]
> > This bug is now occurring to me at 0:19s. instead of at one minute.
> > 
> > I am using a VPN.
> 
> YouTube is pretty quick to blacklist VPN IP addresses in my experience. 

Nope! I can watch hours with Proton, it just takes a bit of dancing to choose the right VPN location (usually Croatia or Cyprus via Proton), get to the UMP fallback, and make certain I never skip outside of the buffered area.

**Comment by SovietGP** [comment]
I have tried to login again and I now have an "No signature available"  message in red. I tried reinstalling the app and plugin and still have this issue.

**Comment by VoxelPrismatic** [comment]
No signature available means that the plugin may not come from the original source. That's fine.

The video stopping at 0:59s is known.

**Comment by VoxelPrismatic** [comment]
> I tried reinstalling the app

This is a needless hassle; it rarely fixes anything. You can clear the cache under App Info > Storage > Clear Cache, that's the best you can do.

**Comment by PGQT** [comment]
> Couldn't get that to work :/

Same here

**Comment by Pyro57000** [comment]
Same, in the USA I do have a VPN but it's a split tunnel, YouTube traffic goes out my normal gateway. I can confrim that without changing networks or anything the YouTube app plays videos just fine. I'm happy to share any logs necessary to help fix this, I would just need instructions for gathering said logs.

**Comment by GeneralProtectionFault** [comment]
I also run into payback stopping at 19 seconds. I didn't use VPN and I don't think I'm IP banned. Obviously, Google always tests their stuff to groups, not all at once. We probably need a standardized procedure to enable some verbose logging when doing bug reports.

**Comment by Zvonimir-FUTO** [comment]
Just to update, with the latest Youtube update, some users are reporting that enabling "**Allow login for video details**" fixes the issue for them. Naturally, you would have to be logged in for it to work. 

If that also doesn't fix your problem, we didn't give up on solving it so don't worry.

**Comment by ThePrivacyPlayer** [comment]
> Just to update, with the latest Youtube update, some users are reporting that enabling "**Allow login for video details**" fixes the issue for them. Naturally, you would have to be logged in for it to work.
> 
> If that also doesn't fix your problem, we didn't give up on solving it so don't worry.

I can confirm this now works flawlessly for me (at least for now ;-) ). Thank you for your work!

And hope you can solve this soon for those who don't use login.

**Comment by dsoto8** [comment]
> > Just to update, with the latest Youtube update, some users are reporting that enabling "**Allow login for video details**" fixes the issue for them. Naturally, you would have to be logged in for it to work.
> > If that also doesn't fix your problem, we didn't give up on solving it so don't worry.
> 
> I can confirm this now works flawlessly for me (at least for now ;-) ). Thank you for your work!
> 
> And hope you can solve this soon for those who don't use login.

This solution did not work for me.

**Comment by dsoto8** [comment]
Wrong button.

**Comment by SovietGP** [comment]
I relogged and reinstalled YouTube plugin to the latest version and everything is working now. No issues with youtube playback or Age restricted playback, thank you for all your hard work 🥰.

**Comment by Pyro57000** [comment]
> Just to update, with the latest Youtube update, some users are reporting that enabling "**Allow login for video details**" fixes the issue for them. Naturally, you would have to be logged in for it to work.
> 
> If that also doesn't fix your problem, we didn't give up on solving it so don't worry.

This did work for me, at least for now. I will report back if it starts happening again.,

**Comment by GeneralProtectionFault** [comment]
It's also working for me this morning with login for video details enabled. However, I think it's still a bug, because that's not supposed to be necessary.  Google could also have rolled back a change from testing means of disrupting the app. So, my 2 cents would be there's still a need for a debug mode or something that can generate a log, if the user permits, that includes all the network info, errors, etc...

**Comment by getabc** [comment]
Plugin upgraded to 223 and UMP sources are working again. Everything seems back to normal.

**Comment by thecatontheceiling** [comment]
Intermittently getting errors like "Failed to get UMP sources, reboot app/plugin", otherwise all seems back to normal now

**Comment by jonathancutting** [comment]
Plays on phone and laptop now, but Fcast streams from YouTube still won't play. I also really don't want to log into YouTube, so if we can get this working without a login, that'd be great.

**Comment by VoxelPrismatic** [comment]
Are you using a VPN, @jonathancutting? If so, there is a "proxy all requests" option in Grayjay settings.

**Comment by jonathancutting** [comment]
Yes, using VPN on both devices. However, problem persists when one or neither of them are connected to the VPN. Enabling "always proxy" doesn't seem to help. The Fcast screen will go blank like it's trying to play, but then it just spins the loading circle forever. Video will play on phone when I disconnect Fcast. YouTube plugin just updated to 224, but that didn't fix it.

**Comment by ThePrivacyPlayer** [comment]
There seems to be a new error appearing for the last few days, and as far as I can see, it only affects longer videos (about 1h+) that I haven't finished watching till the end.

The error I have is `Failed to load UMP, try restarting plugin/app`, and then a bunch of number I don't understand, but might be timestamps.

Unfortunately I'm unable to test this much because I only have two recent long videos that I haven't finished, and all the others appear to work just fine.

I'll leave it here for now, as it concerns UMP stream, but let me know if I should create a separate issue for this.

Cheers!

**Comment by VoxelPrismatic** [comment]
> There seems to be a new error appearing for the last few days, and as far as I can see, it only affects longer videos (about 1h+) that I haven't finished watching till the end.
> 
> The error I have is `Failed to load UMP, try restarting plugin/app`, and then a bunch of number I don't understand, but might be timestamps.
> 
> Unfortunately I'm unable to test this much because I only have two recent long videos that I haven't finished, and all the others appear to work just fine.
> 
> I'll leave it here for now, as it concerns UMP stream, but let me know if I should create a separate issue for this.
> 
> Cheers!

Go into App Info > Storage > Clear Cache. That solved it for me.

**Comment by ThePrivacyPlayer** [comment]
> > There seems to be a new error appearing for the last few days, and as far as I can see, it only affects longer videos (about 1h+) that I haven't finished watching till the end.
> > The error I have is `Failed to load UMP, try restarting plugin/app`, and then a bunch of number I don't understand, but might be timestamps.
> > Unfortunately I'm unable to test this much because I only have two recent long videos that I haven't finished, and all the others appear to work just fine.
> > I'll leave it here for now, as it concerns UMP stream, but let me know if I should create a separate issue for this.
> > Cheers!
> 
> Go into App Info > Storage > Clear Cache. That solved it for me.

Unfortunately, it didn't work for me :-(

**Comment by VoxelPrismatic** [comment]
Did you update to YouTube v224?

**Comment by ThePrivacyPlayer** [comment]
Yes, I'm on v224.

**Comment by VoxelPrismatic** [comment]
Are you using a VPN? Are you logged in? Did you enable the provide login for video details option?

**Comment by ThePrivacyPlayer** [comment]
I get the error whether my VPN is enabled or disabled. And yes to the latter 2 questions.

**Comment by VoxelPrismatic** [comment]
Okay, I can reproduce this. Turn off the "use login for video details" and you should be good.

cc @Kelvin-FUTO

**Comment by barkerd427** [comment]
Everything is also working for me now without needing to login and
downloading UMP.

**Comment by dsoto8** [comment]
I'm still encountering issues:

After updating the YouTube plugin to v224:
• Continue getting Failed to get iOS stream fallback to UMP message.
• Video playback stops abruptly after 4-15minutes of playback.

I have attempted clearing the cache via app settings, which yielded no change.

**Comment by Kelvin-FUTO** [comment]
> I'm still encountering issues:
> 
> After updating the YouTube plugin to v224: • Continue getting Failed to get iOS stream fallback to UMP message. • Video playback stops abruptly after 4-15minutes of playback.
> 
> I have attempted clearing the cache via app settings, which yielded no change.

Fallback to UMP is intended behavior as it means iOS is not available for you, if you dont want the message you can disable "Show Verbose Messages" in the youtube plugin.

4-15 minutes playback sounds unrelated to the issues. However there is still a big issue related to switching networks killing the playback. So if you have these issues on a mobile network, then thats likely the reason. On a single network/ip it should work fine.

**Comment by Zvonimir-FUTO** [comment]
It seems we're all talking about a lot of different issues at this point, and it makes it hard to keep track of whats going on.

Summary:

    Basic playback appears solved for most users in most situations now.
    Download has an issue with the new implementation (issue 

https://github.com/futo-org/grayjay-android/issues/1849 )
Issue with switching networks during playback, which may cause your video playback to stop. (issue

    https://github.com/futo-org/grayjay-android/issues/1854 )
    We are working to fix these issues, but they are unrelated to this particular issue.

Also to repeat, we never disabled iOS source support. Its still prioritized, but they are slowly rolling out that they won't work, and thus Grayjay won't use them if they don't work. Any alignment with the updates is likely because our update was aligned with the wider rollout of this problem (just in time).

**Comment by CoffeeSquirel** [comment]
This is still not working for me. Is no one else having this problem?

It's been like this for well over two months, before it used to load after pressing yes on the prompt, now it's hard stuck on ~1min. It's starting to get VERY annoying.

**Comment by bphd** [comment]
> I have an issue that appears to be related, so sending it here:
> 
> While logged in, with VPN active (but not hopping around), and on either stream or download, the video fails early on. On stream, as in the original post, the video gets to around 1 minute, then stops loading, and becomes unresponsive to the Grayjay player until the video is reopened, and then proceeds to have the same issue. On download, it shows the video size as a few dozen to a few hundred bytes, no matter the size of the video, and throws the following error: `[Youtube] No streams for requesting segment 7 (2384) [3-4]`
> 
> Active: allow loginfallback, allow age restricted, allow controversial restricted
> 
> Also tried just now, but in separated tests (neither worked): activating force experimental ump streams, disabling VPN
> 
> Example of video where the issue was observed in my tests, in case it helps, is Kung Fury (video ID `bS5P_LAqiVg`).
> 
> Youtube extension version 219, Grayjay version 278

It goes fine for the first minute of the video, then it starts to re-download the whole video and re-start where it left off. Inreasing how it does it, first time once for few seconds, then more and more until it does it multiple times per seconds and when it's too much he consider the video is finished whatever the place he was and end playing (happens always at the same time, same frequency, same all)

```typescript
[2026-06-18 00:37:45.169] [INFO] [Program] AppContext.BaseDirectory: /gnu/store/2cl8sh5qbmc1h6q4ibshhxa70in9mmlc-grayjay-12/lib/grayjay/
[2026-06-18 00:37:45.197] [INFO] [Program] Base Directory: /home/$USER/.local/share/Grayjay
[2026-06-18 00:37:45.198] [INFO] [Program] Temporary Directory: /home/$USER/.local/share/Grayjay/temp_files
[2026-06-18 00:37:45.202] [INFO] [ManagedThreadPool] New threadpool created (16)
[2026-06-18 00:37:45.265] [INFO] [Program] Log Level: Warning
[2026-06-18 00:37:45.265] [INFO] [Program] Log file path: /home/$USER/.local/share/Grayjay/log.txt
Log flusher exited with exception: System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at Grayjay.Desktop.POC.Log.<.ctor>b__7_0()
[2026-06-18 00:37:45.590] [WARNING] [StatePlatform] StatePlatform Initialize
[2026-06-18 00:37:46.379] [WARNING] [StateDownloads] Found 0 redundant downloaded files, deleting
[2026-06-18 00:37:47.362] [ERROR] [V8Converter] FAILED TO PARSE VALUE ON String (Value: [undefined])
[2026-06-18 00:37:47.471] [ERROR] [Program] Failed to check updates
Exception: No updater found
Stack Trace:    at Grayjay.ClientServer.Updater.HasUpdate()
   at Grayjay.Desktop.Program.<>c__DisplayClass13_0.<EntryPoint>b__13()
[2026-06-18 00:37:48.062] [ERROR] [StatePlatform] Home failed for plugin [Rumble]: Captcha required
Exception: Captcha required
Stack Trace:    at Grayjay.Engine.GrayjayPlugin.InterceptExceptions[T](Func`1 act, String script)
   at Grayjay.Engine.GrayjayPlugin.EvaluateRawObject(String script, Boolean nullable)
   at Grayjay.Engine.GrayjayPlugin.EvaluatePager[T](String js, Action`1 modifier)
   at Grayjay.Engine.GrayjayPlugin.<GetHome>b__102_0()
   at Grayjay.Engine.GrayjayPlugin.WithIsBusy[T](Func`1 work)
   at Grayjay.Engine.GrayjayPlugin.GetHome()
   at Grayjay.Desktop.POC.Port.States.StatePlatform.<>c__DisplayClass45_0.<GetHomeLazy>b__1(GrayjayPlugin client)
[2026-06-18 00:37:48.216] [ERROR] [StatePlatform] Home failed for plugin [Kick (Beta)]: Request [https://web.kick.com/api/v1/livestreams?page=1&limit=20&sort=featured_home&language=en] failed with code [403] (Line 328, Col 27)
Exception: Request [https://web.kick.com/api/v1/livestreams?page=1&limit=20&sort=featured_home&language=en] failed with code [403] (Line 328, Col 27)

[2026-06-18 00:38:55.123] [ERROR] [HttpProxy] Failed to handle data.
Exception: No streams for requesting segment 13 (Line 5410, Col 10)
   at Grayjay.Engine.Extensions.InvokeV8(IJavaScriptObject obj, PluginConfig config, String method, Object[] args)
   at Grayjay.Engine.Models.Video.Additions.RequestExecutor.ExecuteRequest(String url, Dictionary`2 headers)
   at Grayjay.ClientServer.Controllers.DetailsController.<>c__DisplayClass32_0.<getRequestExecutorProxy>b__0(HttpProxyRequest req)
   at Grayjay.ClientServer.Proxy.HttpProxySession.RunAsync()
   at Grayjay.ClientServer.Proxy.HttpProxySession.<Start>b__7_0()

[2026-06-18 00:39:58.998] [ERROR] [T] An exception occurred in the IPC
Exception: The operation was canceled.
Stack Trace:    at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.IO.Pipes.PipeStream.ReadAsyncCore(Memory`1 destination, CancellationToken cancellationToken)
   at System.IO.Stream.ReadAtLeastAsyncCore(Memory`1 buffer, Int32 minimumBytes, Boolean throwOnEndOfStream, CancellationToken cancellatio
nToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at DotCef.DotCefProcess.<Start>b__28_3()
[2026-06-18 00:39:59.003] [ERROR] [T] An exception occurred while handling UDP result: 'The operation was canceled.':    at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Net.Sockets.SocketReceiveFromResult>.GetResult(Int16 token)
   at System.Net.Sockets.UdpClient.<ReceiveAsync>g__WaitAndWrap|62_0(ValueTask`1 task)
   at FUTO.MDNS.MDNSListener.ReceiveLoopAsync(UdpClient client, CancellationToken cancellationToken) in /home/ubuntu/builds/t2_TNGVSV/0/videostreaming/Grayjay.Desktop/SyncServer/FUTO.MDNS/FUTO.MDNS/MDNSListener.cs:line 268
[2026-06-18 00:39:59.020] [ERROR] [WebSocketEndpoint] Failed to close socket
Exception: The WebSocket is in an invalid state ('Closed') for this operation. Valid states are: 'Open, CloseReceived'
Stack Trace:    at System.Net.WebSockets.WebSocketValidate.ThrowIfInvalidState(WebSocketState currentState, Boolean isDisposed, WebSocketState[] validStates)
   at System.Net.WebSockets.ManagedWebSocket.CloseOutputAsyncCore(WebSocketCloseStatus closeStatus, String statusDescription, CancellationToken cancellationToken)
   at Grayjay.ClientServer.WebSockets.WebSocketClient.Handle()
[2026-06-18 00:39:59.029] [ERROR] [HttpProxy] Failed to accept client.
Exception: Operation canceled
Stack Trace:    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Net.Sockets.Socket>.GetResult(Int16 token)
   at System.Net.Sockets.TcpListener.<AcceptTcpClientAsync>g__WaitAndWrap|32_0(ValueTask`1 task)
   at Grayjay.ClientServer.Proxy.HttpProxy.<Start>b__9_0()
```
